tests/foreman/cli/test_environment.py

```diff
@@ -54,6 +54,108 @@ def test_positive_list_with_name(self):
                 self.assertEqual(len(result), 1)
                 self.assertEqual(result[0]['name'], name)
 
+    @tier2
+    @run_only_on('sat')
+    def test_positive_list_with_org_and_loc(self):
+        """Test Environment List filtering.
+
+        @id: 962c6750-f203-4478-8827-651db208ff92
+
+        @Assert: Results that match both organization and location are returned
+
+        @CaseLevel: Integration
+
+        @BZ: 1337947
+        """
+        # Create 2 envs with the same organization but different locations
+        org = make_org()
+        locs = [make_location() for _ in range(2)]
+        envs = [make_environment({
+            'name': gen_string('alpha'),
+            'organization-ids': org['id'],
+            'location-ids': loc['id'],
+        }) for loc in locs]
+        results = Environment.list({
+            'organization': org['name'],
+        })
+        # List environments for the whole organization
+        self.assertEqual(len(results), 2)
+        self.assertEqual(
+            {env['name'] for env in envs},
+            {result['name'] for result in results}
+        )
+        # List environments with additional location filtering
+        results = Environment.list({
+            'organization': org['name'],
+            'location': locs[0]['name'],
```

Comment:
Original BZ is saying `location-id` is ignored when `organization-id` is provided, so it's not about org/loc names. This test is ok though, no need to drop it, but please implement another one with entity ids.